### ISSUE TITLE
[azure-kinect-sensor-sdk] Run-time crash due to multiple libraries trying to use the same log-file.

### ISSUE THREAD

**Issue by MartinEekGerhardsen** [microsoft/vcpkg#34871]
**Describe the bug**
When using the `k4arecord` library, both `k4a` and `k4arecord` seem to try to log to the same logfile. 

**Environment**
- OS: Windows
- Compiler: MSVC 19.37.32825.0

**To Reproduce**
1. Create new folder with these files: 
CMakeLists.txt
```
add_executable (main "main.cpp" )
find_package(k4a CONFIG REQUIRED)
find_package(k4arecord CONFIG REQUIRED)
target_link_libraries(main PRIVATE k4a::k4a k4a::k4arecord)
``` 
vcpkg.json
```
{ "dependencies": [ "azure-kinect-sensor-sdk" ] }
```

```
#include <iostream>
#include <cstdlib>

#include <k4arecord/playback.hpp>

int main()
{
	std::cout << "K4A_ENABLE_LOG_TO_A_FILE " << std::getenv("K4A_ENABLE_LOG_TO_A_FILE") << '\n';
	std::cout << "K4A_RECORD_ENABLE_LOG_TO_A_FILE " << std::getenv("K4A_RECORD_ENABLE_LOG_TO_A_FILE") << '\n';
	k4a::playback pb = k4a::playback::open("C:\\dev\\testdata\\test.mkv");

	return 0;
}
```
2. Set the following env variables: 
```
$env:K4A_ENABLE_LOG_TO_A_FILE = "0"
$env:K4A_RECORD_ENABLE_LOG_TO_A_FILE = "0"
```
or 
```
$env:K4A_ENABLE_LOG_TO_A_FILE = "k4abase.log"
$env:K4A_RECORD_ENABLE_LOG_TO_A_FILE = "k4arecord.log"
```
3. Set up cmake
```
cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake
cmake --build .\build\ --config Debug
.\build\Debug\test.exe
```

**Expected behavior**
The program shows the current env varables (to ensure that I know these are set) and then the program executes without issue. 

**Failure logs**
If I set the env variables to "0", I get: 
```
ERROR: Logger unable to open log file "k4a.log".
```
If I set the env variables to the two separate log files, I get: 
```
ERROR: Logger unable to open log file "k4abase.log"
```
Furthermore, I get this debug error:
![bilde](https://github.com/microsoft/vcpkg/assets/89084130/8a79e9c6-c47d-437d-9034-52447f8a37ae)

**Additional context**
I believe this is an issue with vcpkg, as I have done this versions of the azure kinect sensor sdk installed to this system without vcpkg. When I use these, I don't get the same error. However, it would be much simpler for me to use vcpkg as I also need the pcl library, which is extremely difficult to install without vcpkg.

**Comment by FrankXie05** [comment]
@MartinEekGerhardsen Thanks for posting this issue. Could you please use the full path of this file?
```
$env:K4A_ENABLE_LOG_TO_A_FILE = "k4abase.log"
$env:K4A_RECORD_ENABLE_LOG_TO_A_FILE = "k4arecord.log"
```

**Comment by MartinEekGerhardsen** [comment]
@FrankXie05 Thank you for the reply! I am not sure I understood what you wanted me to do but I tried setting the full path and got this result: 

![bilde](https://github.com/microsoft/vcpkg/assets/89084130/a4f22b7c-c801-41a4-ab96-c30cd02bacfd)

As you can see, the same error is produced, and both log files are created in the directory I'm executing the code from.

**Comment by MartinEekGerhardsen** [comment]
@FrankXie05 Is there any other information I can provide?

**Comment by FrankXie05** [comment]
@MartinEekGerhardsen Sorry for the late reply, could you please provide a valid Azure Kinect recording file, such as your `test.mkv`? :)

**Comment by FrankXie05** [comment]
Could you please provide your `test.mkv`?(Make sure it conforms to the format of Azure Kinect) :)

**Comment by MartinEekGerhardsen** [comment]
I have uploaded it here: https://fastupload.io/F3VfZPxSfjReXGQ/file The file should be fine, as I recorded it with the Azure Kinect record command.

**Comment by MartinEekGerhardsen** [comment]
Hello @FrankXie05, have you had any time to look at this problem? I haven't been able to figure anything more out on my end 😅

**Comment by MartinEekGerhardsen** [comment]
ping @FrankXie05

**Comment by FrankXie05** [comment]
@MartinEekGerhardsen  Sorry for the late reply. It looks like this is a functional problem of `azure-kinect-sensor-sdk`, not a problem of vcpkg. I'm not an expert on `azure-kinect-sensor-sdk`, you can submit an issue to https://github.com/microsoft/Azure-Kinect-Sensor-SDK, I think you can find the answer there. :)

**Comment by MartinEekGerhardsen** [comment]
@FrankXie05 Thanks for the help! I'll do that, and reference this issue. I'll also keep this issue open for now. The reason why I thought this issue could be related to `vcpkg` is because I was able to get the code to run when I used system installs instead of `vcpkg`, but I'd obviously prefer having everything defined using `vcpkg`. [Here is a link](https://github.com/microsoft/Azure-Kinect-Sensor-SDK/issues/1962) to the new issue. If I figure something out, I'll update here.

**Comment by MartinEekGerhardsen** [comment]
Yeah that's what I ended up concluding as well. I have no idea why the k4arecord library calls tries to use the same logger, as from the docs it seemed like it should have its own separate logger. I think the primary problem is that two different libraries tries to access the same file, so the two cases in k4a.lib is fine, though when adding k4arecord.lib, the error is triggered. 

No idea how k4aviewer avoids this though... Could be interesting to investigate at some point. 

Still, as it seems like azure kinect has been discontinued, I'm not sure any updates will ever come from this...

**Comment by thundercarrot** [comment]
The problem doesn't appear when running with k4aviewer compiled with the sdk not under vcpkg. 

I think the issue is that when compiled with vcpkg, azure-kinect-sensor-sdk uses the version of spdlog that vcpkg pulls down. That seems to be newer than the one used by the vanilla k4a repo, and evidently includes a breaking change.

**Comment by MartinEekGerhardsen** [comment]
Yeah that was why I initially posted this issue here, as the same code example worked with my custom version of Azure-Kintect-Sensor-SDK.

I notice that Azure-Kintect-Sensor-SDK links to spdlog 0.17.0. Now I have no idea if there are any critical vulnerabilities in the code since 0.17.0 (spdlog is at 1.13.0 atm), but if this works, perhaps we can just depend on spdlog version 0.17.0? `

**Comment by martiege** [comment]
@FrankXie05 any updates on this?

**Comment by MartinEekGerhardsen** [comment]
@FrankXie05 any updates on this?

**Comment by FrankXie05** [comment]
It looks like the upstream is not responding.🧐
